service/src/main/kotlin/fi/espoo/evaka/jamix/JamixService.kt

```diff
@@ -106,11 +106,15 @@ class JamixService(
         if (client == null) error("Cannot sync diet list: JamixEnv is not configured")
         val warningEmails =
             fetchAndUpdateJamixDiets(client, db, clock, emailEnv.sender(Language.fi))
-        fetchAndUpdateJamixTextures(client, db)
-        warningEmails.forEach { emailClient.send(it) }
+        try {
+            fetchAndUpdateJamixTextures(client, db)
+        } finally {
+            warningEmails.forEach { emailClient.send(it) }
```

Comment:
Now any error in the texture sync is silently ignored, so that probably isn't what we want. Also this still has plenty of error scenarios that aren't properly handled and exactly-once-delivery can not be guaranteed with this approach. For that each email needs to be it's own separately retryable job, and those need to be planned within the same transaction that `fetchAndUpdateJamixDiets` uses. 

On the other hand this is a feature that triggers very rarely, so maybe it doesn't have to be bullet proof. Then again, if it ever does trigger it is quite crucial. So I don't know, could quickly discuss this after daily with others.

Reply:
I added error logging and a try-catc around `emailClient.send(it)`

Reply:
Ok, maybe it's good enough for now. I'll add some ticket on the bottom of tech backlog to make it less of a hax later if there's time.